src/astx/literals/collections.py

```diff
@@ -0,0 +1,157 @@
+"""ASTX literal representations of collection types."""
+
+from __future__ import annotations
+
+from typing import Dict, List, Set, Tuple
+
+from public import public
+
+from astx.base import AST, NO_SOURCE_LOCATION, SourceLocation
+from astx.literals.base import Literal
+from astx.tools.typing import typechecked
+from astx.types.base import AnyType
+from astx.types.collections import ListType, MapType, SetType, TupleType
+
+
+@public
+@typechecked
+class LiteralList(Literal):
+    """Literal representation of a list."""
+
+    elements: List[Literal]
+
+    def __init__(
+        self,
+        elements: List[Literal],
+        loc: SourceLocation = NO_SOURCE_LOCATION,
+    ) -> None:
+        super().__init__(loc)
+        self.elements = elements
+
+        element_type_classes = {type(elem.type_) for elem in elements}
+
+        if len(element_type_classes) == 1:
+            element_type_class = element_type_classes.pop()
+            element_type: AST = element_type_class()
+        else:
+            element_type = AnyType()
+
+        self.type_ = ListType(element_type)
+        self.loc = loc
+
+    def __str__(self) -> str:
+        """Return a string representation of the list."""
+        return f"LiteralList({self.elements})"
```

Comment:
maybe the best way to do it .. but something like this:

```suggestion
        return f"LiteralList[{self.type_}]"
```

it will be shown as:
> LiteralList[ListType]

so maybe it is not perfect .. but acceptable 

the best output would be something like:

> LiteralList[Int32, Float32]

Reply:
a friendly reminder about this comment.